client/src/utils/track.ts

```diff
@@ -78,46 +79,106 @@ export const track = (event: AnalyticsEvent): Promise<unknown> | undefined => {
+    // @ts-ignore
+    s.pageURL = window.location.href;
     // @ts-ignore
     s.t();
   }
+  firstPageOfVisit = false;
 };
 
 export const trackPageFetchException = (): void => {
   // @ts-ignore
-  if (Build.isBrowser && typeof(s) != "undefined") {
+  if (Build.isBrowser && typeof s != "undefined") {
+    // @ts-ignore
+    s.linkTrackVars =
+      "prop39,prop41,prop50,prop61,prop62,eVar39,eVar41,eVar50,eVar61,eVar62,eVar69";
     // @ts-ignore
     s.tl(true, "o", "page fetch exception");
   }
 };
 
-export const trackExternalLink = (): void => {
+export const trackExternalLink = (hrefTo: string): void => {
   // @ts-ignore
-  if (Build.isBrowser && typeof(s) != "undefined") {
+  if (Build.isBrowser && typeof s != "undefined") {
     // @ts-ignore
-    s.tl(true, "e");
+    s.tl(true, "e", hrefTo);
   }
 };
 
 export const setSearchQuery = (query: string): void => {
   // @ts-ignore
-  if (Build.isBrowser && typeof(s) != "undefined") {
+  if (Build.isBrowser && typeof s != "undefined") {
     // @ts-ignore
     s.eVar26 = query;
   }
 };
 
-export const trackSearchResult = (resultCount: number): void => {
+const triggerNoSearchResults = (query: string): void => {
   // @ts-ignore
-  if (Build.isBrowser && typeof(s) != "undefined") {
-    // @ts-ignore
-    s.linkTrackVars = "eVar26,eVar27";
+  const queryBackup: string = s.eVar26;
+  // @ts-ignore
+  const resultCountBackup: number = s.eVar27;
+
+  // @ts-ignore
+  s.eVar26 = query;
+  // @ts-ignore
+  s.eVar27 = "0"; // If it's the number 0, the variable won't be sent
+  // @ts-ignore
+  s.linkTrackVars =
+    "prop39,prop41,prop50,prop61,prop62,eVar26,eVar27,eVar39,eVar41,eVar50,eVar61,eVar62,eVar69,events";
+  // @ts-ignore
+  s.linkTrackEvents = "event2";
+  // @ts-ignore
+  s.events = "event2";
+  // @ts-ignore
+  s.tl(true, "o", "internal search");
+
+  // @ts-ignore
+  s.eVar26 = queryBackup;
+  // @ts-ignore
+  s.eVar27 = resultCountBackup;
+};
+
+let noResultsTimeout: NodeJS.Timeout;
+export const setSearchResultCount = (resultCount: number): void => {
+  if (Build.isBrowser) {
     // @ts-ignore
     s.eVar27 = resultCount;
     // @ts-ignore
     s.events = resultCount === 0 ? "event1" : "event2";
+
+    if (resultCount === 0) {
+      if (noResultsTimeout) {
+        clearTimeout(noResultsTimeout);
+      }
+      noResultsTimeout = setTimeout(
+        // @ts-ignore
+        triggerNoSearchResults.bind(null, s.eVar26),
+        1000,
+      );
+    }
+  }
+};
+
+export const trackSearchQuery = (
+  _input,
+  _event,
+  suggestion,
+  _datasetNumber,
+  _context,
+): void => {
+  if (Build.isBrowser) {
+    // @ts-ignore
+    s.linkTrackVars =
+      "prop39,prop41,prop50,prop61,prop62,eVar26,eVar27,eVar39,eVar41,eVar50,eVar61,eVar62,eVar69,events";
+    // @ts-ignore
+    s.linkTrackEvents = "event1";
+    // @ts-ignore
+    s.events = "event1";
     // @ts-ignore
     s.tl(true, "o", "internal search");
   }
+  window.location.assign(suggestion.url);
```

Comment:
Why this?

Reply:
We're overriding the default behavior when clicking on a search result in order to send the analytics event.  When we're done tracking, we need to also implement the default behavior: sending the user to the page they clicked on, which wouldn't happen at all if we didn't manually do it.

Do you think this deserves a comment explaining it?

Reply:
Nah, I just wanted clarity is all on it so thank you.